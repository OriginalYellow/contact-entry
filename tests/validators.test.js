const validContact = require('./valid-contact.mock.json')
const invalidContact = require('./invalid-contact.mock.json')
const { validate, errors } = require('partial.lenses.validation')

const { Model, validateContact } = require('../validators.js')

it('should consider the name object to be valid', () => {
  const result = errors(Model.name, validContact.name)
  expect(result).toBeUndefined()
})

it('should return messages saying that "first" and "last" are not strings, and that "middle" is nil or empty', () => {
  const invalidName = { middle: '', last: 3 }
  const result = errors(Model.name, invalidName)
  expect(result).toMatchSnapshot()
})

it('should return a message saying that invalidName is not an object', () => {
  const invalidName = 'asdf'
  const result = errors(Model.name, invalidName)
  expect(result).toMatchSnapshot()
})

it('should consider the address object to be valid', () => {
  const result = errors(Model.address, validContact.address)
  expect(result).toBeUndefined()
})

it('should return messages describing the different errors in the invalid address object', () => {
  const invalidAddress = {
    street: '',
    city: 'cool city',
    state: {},
    zip: 23452
  }
  const result = errors(Model.address, invalidAddress)
  expect(result).toMatchSnapshot()
})

it('should return a message saying that invalidAddress is not an object', () => {
  const invalidAddress = 3
  const result = errors(Model.address, invalidAddress)
  expect(result).toMatchSnapshot()
})

it('should consider the phone array to be valid', () => {
  const result = errors(Model.phone, validContact.phone)
  expect(result).toBeUndefined()
})

it('should return an array that indicates which element in the invalid phone array is invalid, and why', () => {
  invalidPhoneArr = [
    {
      number: '13434243',
      type: 'work'
    },
    {
      number: '123123123123',
      type: 'banana'
    },
    {
      number: '13434243',
      type: 'home'
    },
  ]

  const result = errors(Model.phone, invalidPhoneArr)
  expect(result).toMatchSnapshot()
})

it('should consider the valid contact to be valid', () => {
  const result = validateContact(validContact)
  expect(result).toBeUndefined()
})

it('should return error messages describing the different errors in the invalid contact object', () => {
  const result = validateContact(invalidContact)
  expect(result).toMatchSnapshot()
})
