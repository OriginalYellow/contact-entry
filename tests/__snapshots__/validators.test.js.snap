// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should return a message saying that invalidAddress is not an object 1`] = `"value is not an object"`;

exports[`should return a message saying that invalidName is not an object 1`] = `"value is not an object"`;

exports[`should return an array that indicates which element in the invalid phone array is invalid, and why 1`] = `
Array [
  null,
  Object {
    "type": "\\"type\\" is not \\"mobile\\", \\"work\\", or \\"home\\"",
  },
  null,
]
`;

exports[`should return error messages describing the different errors in the invalid contact object 1`] = `
Object {
  "address": Object {
    "state": "value is not a string",
    "street": "value is nil or empty",
    "zip": "value is not a string",
  },
  "name": Object {
    "first": "value is not a string",
    "last": "value is not a string",
    "middle": "value is nil or empty",
  },
  "phone": Array [
    null,
    Object {
      "type": "\\"type\\" is not \\"mobile\\", \\"work\\", or \\"home\\"",
    },
    null,
  ],
}
`;

exports[`should return messages describing the different errors in the invalid address object 1`] = `
Object {
  "state": "value is not a string",
  "street": "value is nil or empty",
  "zip": "value is not a string",
}
`;

exports[`should return messages saying that "first" and "last" are not strings, and that "middle" is nil or empty 1`] = `
Object {
  "first": "value is not a string",
  "last": "value is not a string",
  "middle": "value is nil or empty",
}
`;
